### ISSUE TITLE
Notification progress bar does not respect "chapter track" setting

### ISSUE THREAD

**Issue by DDriggs00** [advplyr/audiobookshelf-app#239]
### Steps to reproduce

1. Have an audiobook with chapters
2. Play the book, and turn on the "chapter track" setting
3. Exit the app and expand the notification to reveal the progress bar

### Expected behaviour
- The progress bar covers only the current chapter when chapter tracks are enabled

### Actual behaviour
- The progress bar covers the entire book, not just the current chapter

### Environment data

Audiobookshelf Version:

- [X] Android App?
- [ ] iOS App?

#### Android Issue
Android version: Oxygen OS 11.0.7.1.HD65AA

Device model: OnePlus 7T

Stock or customized system: Stock

**Comment by advplyr** [comment]
Are you saying it doesn't work when you close and re-open the app? Or it doesn't work at all?

Are you able to see the chapters in the web app to make sure they are recognized by abs?

Do you have any audiobooks that do work as expected?

**Comment by DDriggs00** [comment]
It doesn't work at all. The in-app progress bar (with both collapsed and full-screen players) works as it should, showing progress for the current chapter. It is only the notification (seen when outside the app) that doesn't respect the setting. The web app shows the chapters. All audiobooks with chapters are affected. [Screenshots](https://imgur.com/a/6lby7R0)

**Comment by advplyr** [comment]
Oh I see what you mean. The notification is getting passed in the duration of the currently playing audio file I believe. Are you using a single audio file with chapters?
I think I have an idea on how this could be fixed. It may also be worthwhile showing the chapter title in place of the book title and the book title in place of the author.

**Comment by DDriggs00** [comment]
Yes, I think all the books I've tested have been single-file m4b with internal chapter marks.

**Comment by ronaldheft** [comment]
I love this idea, and will look to implement this on the iOS side.

**Comment by advplyr** [comment]
I added an explanation for why this happens here https://github.com/advplyr/audiobookshelf-app/issues/729#issuecomment-1575082870 for future reference

**Comment by advplyr** [comment]
This has been added for iOS in [v0.9.71-beta](https://github.com/advplyr/audiobookshelf-app/releases/tag/v0.9.71-beta) so this is Android specific now

**Comment by TheDogg** [comment]
Any updates on this issue? Still not fixed on Android. Would really appreciate it, having the whole file in the notification is kind of annoying.

Thanks

**Comment by quartertone** [comment]
Yes, please implement this on Android!

It's kind of a necessary feature since it's very easy to accidentally touch the progress bar which would completely take you out of place.

Implementing https://github.com/advplyr/audiobookshelf-app/issues/978 (ability to turn off progress bar) could be a hacky fix for this scenario, but personally I would prefer having the per-chapter progress bar.

Edit: I just found the "Allow position seeking on media notification controls" toggle in settings. This would prevent accidental seeking, but it would still be nice to be able to see per-chapter progress.
